src/controllers/Game.ts

```diff
@@ -1,101 +1,78 @@
-import { Client, Models, Packets, TAEvents } from "tournament-assistant-client";
-import UIWebSocketManager from "./UIWebSocketManager";
-import fetch from "node-fetch";
-
-interface PlayerUser {
-    name: string,
-    type: Models.User.ClientTypes,
-    user_id: string,
-    guid: string,
-    stream_delay_ms: number,
-    stream_sync_start_ms: number,
-}
-
-interface PlayerScore {
-    user_id: string | undefined
-    user_guid: string
-    score: number
-    accuracy: number
-    combo: number
-    notesMissed: number
-    badCuts: number
-    bombHits: number
-    wallHits: number
-    maxCombo: number
-    lhAvg: number[]
-    lhBadCut: number
-    lhHits: number
-    lhMiss: number
-    rhAvg: number[]
-    rhBadCut: number
-    rhHits: number
-    rhMiss: number
-    totalMisses: number
-}
-
+import got from 'got';
+import UIWebSocket from './UIWebSocket.js';
+import {
+    Models,
+    Packets,
+} from 'tournament-assistant-client';
+import { logger } from '../main.js';
+import { ScoreSaberPlayerInfo } from '../types/ScoreSaber.js';
+import { TAPlayerInfo, TAScore } from '../types/TA.js';
+
+// ? Early deprecation notice: This class will be refactored out in favor of the TA Client's State object.
+// ? Perferably, we will have our own StateController class that will handle all of *our* state management.
 export default class Game {
 
     users: Map<string, Models.User>;
     matches: Map<string, Models.Match>;
-    scores: Map<string, PlayerScore>;
+    scores: Map<string, TAScore>;
     currentMatch: string;
-    scoresabers: Map<string, any>;
+    scoresabers: Map<string, unknown>;
 
-    constructor(private uiSocket: UIWebSocketManager) {
+    constructor(private uiSocket: UIWebSocket) {
         this.users = new Map();
         this.matches = new Map();
         this.scores = new Map();
         this.scoresabers = new Map();
         this.currentMatch = "";
     }
 
-    getCurrentMatchPlayers(): PlayerUser[] {
+    getCurrentMatchPlayers(): TAPlayerInfo[] {
         const playerGuids = this.matches.get(this.currentMatch)?.associated_users ?? [];
-        return playerGuids.map(guid => this.playerUsers.find(u => u.guid === guid)).filter(Boolean) as PlayerUser[];
+        return playerGuids.map(guid => this.playerUsers.find(u => u.guid === guid)).filter(Boolean) as TAPlayerInfo[];
     }
 
-    onGameStateUpdate() {
+    async onGameStateUpdate() {
         const players = this.getCurrentMatchPlayers();
-        this.uiSocket.sendToUI("game-state-update", { 
+        this.uiSocket.sendToUI("game-state-update", {
             match: this.matches.get(this.currentMatch)?.toObject(),
             players,
         });
 
         const ids = players.map(p => p.user_id);
-        this.updateScoresabers(ids).then(scoresabers => {
-            console.log(scoresabers);
+
+        await this.updateScoresabers(ids).then(scoresabers => {
```

Comment:
You don need .then() anymore when using await

Reply:
👍

Reply:
you can resolve these discussion as you see fit :)